apps/SageAccountWeb/src/components/ResetTwoFactorAuth.tsx

```diff
@@ -0,0 +1,164 @@
+import { Alert, Box, Button, Typography } from '@mui/material'
+import React, { useMemo, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import {
+  displayToast,
+  StandaloneLoginForm,
+  SynapseQueries,
+  useApplicationSessionContext,
+} from 'synapse-react-client'
+import {
+  TwoFactorAuthErrorResponse,
+  TwoFactorAuthResetToken,
+} from '@sage-bionetworks/synapse-types'
+import { getSearchParam, hexDecodeAndDeserialize } from '../URLUtils'
+import { BackButton } from './BackButton'
+import { LeftRightPanel } from './LeftRightPanel'
+import { SourceAppLogo } from './SourceApp'
+import { RESET_2FA_SIGNED_TOKEN_PARAM } from '../Constants'
+
+export function ResetTwoFactorAuth() {
```

Comment:
New component displayed on the `/reset2FA` route.

It handles:
- Getting the twoFaResetToken out of the search param, and deserializing it
- Prompting the user to log in (to get a twoFaToken)
- Prompting the user to confirm disabling 2FA
- Sending the twoFaResetToken and twoFaToken to the backend to disable 2FA when the user confirms their choice.